src/components/Login.js

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable no-undef */
+import {initializeApp} from "firebase/app";
+import React, {useState} from "react";
+import {useNavigate} from "react-router-dom";
+import {
+  getAuth,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import {Button} from "../styles/Profile.module";
+import {getFirestore, doc, setDoc, onSnapshot} from "firebase/firestore";
+import {LoginWrapper} from "../styles/Login.module";
+
+const firebaseConfig = {
+  apiKey: process.env.REACT_APP_FIREBASE_APIKEY,
+  authDomain: process.env.REACT_APP_FIREBASE_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECTID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGEBUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGINGSENDERID,
+  appId: process.env.REACT_APP_FIREBASE_APPID,
+  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENTID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const auth = getAuth();
+const db = getFirestore(app);
+
+function Login(props) {
+  const SIGN_IN = "sign_in";
+  const SIGN_UP = "sign_up";
+
+  const [isShowSignWhat, setIsShowSignWhat] = useState();
+  const [userEmail, setUserEmail] = useState("");
+  const [userPassword, setUserPassword] = useState("");
+  const [userName, setUserName] = useState("");
+  const [userPhoto, setUserPhoto] = useState("");
+  const [userRole, setUserRole] = useState("");
+  const [userLink, setUserLink] = useState("");
+
+  const redirect = useNavigate();
+
+  const showSingIn = () => {
+    setIsShowSignWhat(SIGN_IN);
+  };
+
+  const showSingUp = () => {
+    setIsShowSignWhat(SIGN_UP);
+  };
+
+  const signUp = () => {
+    createUserWithEmailAndPassword(auth, userEmail, userPassword)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        writeUserData(user.uid);
+        getUserData(user.uid);
+      })
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.log("errorCode", errorCode);
+        console.log("errorMessage", errorMessage);
+
+        switch (errorCode) {
+          case "auth/wrong-password":
+            alert("密碼錯誤喔");
+            break;
+          case "auth/user-not-found":
+            alert("帳號不存在喔");
+            break;
+          case "auth/weak-password":
+            alert("密碼需至少六碼喔");
+            break;
+          default:
+            alert("登入失敗QQ");
+        }
```

Comment:
Please mange error code with another constant file

Reply:
Have them arranged in const.js.